src/OmniSharp.Cake/Services/RequestHandlers/Navigation/FindSymbolsHandler.cs

```diff
@@ -21,12 +21,18 @@ public FindSymbolsHandler(OmniSharpWorkspace workspace)
 
         public override Task<QuickFixResponse> Handle(FindSymbolsRequest request)
         {
+            if (request != null && request.Filter != null && request.Filter.Length < request.MinFilterLength)
+            {
+                return Task.FromResult(new QuickFixResponse { QuickFixes = Array.Empty<QuickFix>() });
+            }
+
             Func<string, bool> isMatch =
                 candidate => request != null
                 ? candidate.IsValidCompletionFor(request.Filter)
                 : true;
 
-            return Workspace.CurrentSolution.FindSymbols(isMatch, LanguageNames.Cake);
+            int maxItemsToReturn = request == null ? 0 : request.MaxItemsToReturn;
```

Comment:
Request is already not null here

Reply:
Hmm, actually when invoked from UTs it WAS null and logic of "isMatch" checks for null as well. Are you implying that IRequestHandler.Handle can never receive a null?

Reply:
Don't you check for not-null on line 24?

Reply:
The previous check is more broad though: (request != null && request.Filter != null && request.Filter.Length < request.MinFilterLength. It will still allow to request to be null when maxItemsToReturn  is calculated. I did this to prevent changing existing behavior of the handler when MinFilterLength is not provided or doesn't need to be applied. Would you suggest to always return empty result when request == null? Should this be done then across the board for all requests?

Reply:
Right--reading comprehension :)